### ISSUE TITLE
Installation issue: root

### ISSUE THREAD

**Issue by davehadley** [LLNL/spack#23029]
Spec: `root@6.22.00 +python` does not correctly build pyroot bindings

### Steps to reproduce the issue

```shell
spack install root@6.22.00 +python
spack load root@6.22.00 +python
python -c "import ROOT"
```
produces the error message:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'ROOT'
```

I do not get this problem building `root@6.16.00`.

### Information on your system

* **Spack:** 0.16.1-2167-982d3ab
* **Python:** 3.6.8
* **Platform:** linux-centos7-skylake
* **Concretizer:** original

### Additional information

I believe that the issue is caused by the lines:
```
        # Some special features
        if self.spec.satisfies('@6.20:'):
            options.append(define_from_variant('pyroot', 'python'))
```

ROOT recently changed the build option from `python` to `pyroot`.
These lines implement that change for `>=6.20`.
However, examining the ROOT source code I see the new pyroot build command was introduced inbetween:
[v6-20-00](https://github.com/root-project/root/blob/v6-20-00/cmake/modules/RootBuildOptions.cmake) and
[v6-20-02](https://github.com/root-project/root/blob/v6-20-02/cmake/modules/RootBuildOptions.cmake)

Thus I believe that these lines should be something like:
```
        # Some special features
        if self.spec.satisfies('@6.20.02:'):
            options.append(define_from_variant('pyroot', 'python'))
```

spack maintainers: @chissg, @HadrienG2 @drbenmorgan @vvolkl

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by vvolkl** [comment]
Thanks @davehadley for the report, and providing the fix already! Your proposed change looks reasonable to me. If you can open a PR with the changed line, I'd say, just go ahead.

**Comment by davehadley** [comment]
Yes I can put in a PR once I test that this fix builds successfully.

**Comment by davehadley** [comment]
I was able to build with the proposed change and now pyroot works as expected. Fix is in PR #23038
